app/main.py

```diff
@@ -1,34 +1,64 @@
-class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+from typing import List, Tuple, Dict
+
 
+class Deck:
+    def __init__(self, row: int, column: int,
+                 is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: Tuple[int, int], end: Tuple[int, int],
+                 is_drowned: bool = False) -> None:
+        self.decks: List[Deck] = []
+        self.is_drowned = is_drowned
+        self.create_decks(start, end)
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def create_decks(self, start: Tuple[int, int],
+                     end: Tuple[int, int]) -> None:
+        start_row, start_col = start
+        end_row, end_col = end
+        if start_row == end_row:
+            for col in range(start_col, end_col + 1):
+                self.decks.append(Deck(start_row, col))
+        elif start_col == end_col:
+            for row in range(start_row, end_row + 1):
+                self.decks.append(Deck(row, start_col))
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        return None
 
+    def fire(self, row: int, column: int) -> bool:
+        deck = self.get_deck(row, column)
+        if deck and deck.is_alive:
+            deck.is_alive = False
+        if all(not d.is_alive for d in self.decks):
+            self.is_drowned = True
+            return True
+        return False
```

Comment:
This code block in the `fire` method has redundant `return` statements. Here, if a deck is found and it is alive, you set `deck.is_alive` to `False`, but you don't return any value. Then, you check if all decks are not alive and if so, you set `self.is_drowned` to `True` and return `True`. This can be simplified by removing the redundant `return` statements. Please refer to the checklist item 2 for a good example.